src/libostree/ostree-sysroot.c

```diff
@@ -813,6 +815,17 @@ ostree_sysroot_load_if_changed (OstreeSysroot  *self,
                                cancellable, error))
     return FALSE;
 
+  /* Determine whether we're "physical" or not, the first time we initialize */
+  if (!self->loaded)
+    {
+      if (fstatat (self->sysroot_fd, "sysroot", &stbuf, 0) < 0)
```

Comment:
(BTW I'm trying to avoid any path comparisons because part of the idea here is that one can create a sysroot from a directory fd)